lib/types/lvm_vg.nix

```diff
@@ -76,9 +88,12 @@
             if ! lvdisplay '${config.name}/${lv.name}'; then
               lvcreate \
                 --yes \
-                ${if lib.hasInfix "%" lv.size then "-l" else "-L"} ${lv.size} \
+                ${if (lv.lvm_type != "thinlv") then
+                  (if lib.hasInfix "%" lv.size then "-l" else "-L")
+                  else "-V"} ${lv.size} \
                 -n ${lv.name} \
-                ${lib.optionalString (lv.lvm_type != null) "--type=${lv.lvm_type}"} \
+                ${lib.optionalString (lv.lvm_type == "thinlv") "--thinpool=${lv.pool}"} \
```

Comment:
Can you add an example to examples for thinlv or extend an existing one so that we can have an integration test for it?

Reply:
I added a fairly simple example and test based on one of the other lvm ones.

Reply:
does the thinpool need to exist before? maybe we need to do some ordering in that case? or is the thinpool always `size = "100%FREE` ?

Reply:
The thinpool does need to exist before any of its thin LVs which is taken care of by the priority option's default value.
https://github.com/nix-community/disko/blob/de29130286bbad9e17ea71f6bd99a272a1ec5c75/lib/types/lvm_vg.nix#L30